src/api/cpp/backend/backend_engine.h

```diff
@@ -26,87 +27,89 @@
 class nixlBackendEngine {
     private:
         // Members that cannot be modified by a child backend and parent bookkeep
-        nixl_backend_t   backendType;
-        nixl_b_params_t* customParams;
+        nixl_backend_t  backendType;
+        nixl_b_params_t customParams;
 
     protected:
         // Members that can be accessed by the child (localAgent cannot be modified)
-        bool              initErr;
+        bool              initErr = false;
         const std::string localAgent;
 
-        nixl_status_t setInitParam(const std::string &key, const std::string &value) {
-            if (customParams->count(key)==0) {
-                (*customParams)[key] = value;
+        [[nodiscard]] nixl_status_t setInitParam(const std::string &key, const std::string &value) {
+            if (customParams.count(key)==0) {
+	        customParams[key] = value;
                 return NIXL_SUCCESS;
             } else {
                 return NIXL_ERR_NOT_ALLOWED;
             }
         }
 
-        nixl_status_t getInitParam(const std::string &key, std::string &value) {
-            if (customParams->count(key)==0) {
+        [[nodiscard]] nixl_status_t getInitParam(const std::string &key, std::string &value) {
+            if (customParams.count(key)==0) {
                 return NIXL_ERR_INVALID_PARAM;
             } else {
-                value = (*customParams)[key];
+	        value = customParams.at(key);
```

Comment:
nit: looking up the iterator only once, checking whether it is equal to customParams.end() and, if not, assigning iterator->second to value would be more elegant (and would also allow you to add const modifier to this function I assume).